### ISSUE TITLE
Provide man page in source tarball or as a release asset

### ISSUE THREAD

**Issue by Biswa96** [nuspell/nuspell#137]
- Change request or feature request

# Change/feature request

Would it be possible to add man page in source tarball or release asset after generating it from markdown file? pandoc and haskell toolchain is not available for many systems. See the list of distributions and systems here https://www.haskell.org/ghc/distribution_packages.html and https://www.haskell.org/ghc/download_ghc_9_10_1.html. It can be quite inconvenient to download a large program like Pandoc just to convert a single man page. It could be done with CI or manually.

**Comment by dimztimz** [comment]
> Would it be possible to add man page in source tarball or release asset after generating it from markdown file?

No. I might change to a different generator some day in some future. For now, just build with manpage disabled.

> pandoc and haskell toolchain is not available for many systems.

I am aware of this. But it is not really many. MSYS2 and which else? What should be done to put Haskell into MSYS?

> It can be quite inconvenient to download a large program like Pandoc just to convert a single man page. It could be done with CI or manually.

This is not a good argument. Should we all stop using dependencies?

**Comment by Biswa96** [comment]
> But it is not really many. MSYS2 and which else? What should be done to put Haskell into MSYS?

My issue is not specific to msys2 project but others also including msys2. For example, Alpine Linux has two architecture for ghc https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/ghc/APKBUILD

**Comment by badshah400** [comment]
For openSUSE Tumbleweed/Leap also we would like this very much. As nuspell has become a core component of our basic desktop GNOME packages, it is included in the Ring1 tier which goes through advanced automated build, install, and QA testing. Introducing too many dependencies into Ring1 — as pulling in `pandoc` would — will obviously slow down Ring1 builds and tests and we would like to avoid that as much as possible. For now, we do disable the man file generation for nuspell, but if it is pre-generated as part of the source `tar.gz` instead of being regenerated at every build, we could easily package it and provide this nice help file to our users.

See comments at <https://build.opensuse.org/request/show/1188463> for more context.

Many thanks for your great library.

**Comment by dimztimz** [comment]
@badshah400 can you explain more on this stuff you call Ring1? Is there a list of software that is part of it? I have not see another distro with such limitation.

**Comment by badshah400** [comment]
For openSUSE Tumbleweed, Ring1 packages are those that constitute the product DVD and main desktop environments supported by the distribution, such as GNOME, KDE, Xfce, etc. Packages in Ring 1 undergo substantially more (automated) testing, with all tests required to pass before a new snapshot of the distribution may be released. More the number of packages included in this ring, more the time taken to build them all, test them all and so on before releasing a new Tumbleweed snapshot. Therefore, one tries to avoid a large set of additional build dependencies for packages already in Ring1, to avoid ballooning the Ring1 build and test durations and slowing down the release of new snapshots.

You call this a limitation, but I think it is a feature that allows TW to only release when at least a core set of packages are not broken by some update.

In any case, the man page is static, version tied (hopefully), and does not depend on the build machine or configuration. So, in my opinion, it should not be a problem building it as part of `make dist` or equivalent and supplying it as part of the release tarball. Just my 2¢.

Thanks.

**Comment by badshah400** [comment]
Sorry, missed the list of packages in Ring1. I _think_ this is it: <https://build.opensuse.org/project/show/openSUSE:Factory:Rings:1-MinimalX>
